src/utils.jl

```diff
@@ -0,0 +1,28 @@
+function trycopy(data)
+    # Not all objects support `copy`, but we should use it to improve
+    # performance if possible.
+    try
+        copy(data)
+    catch
+        deepcopy(data)
+    end
```

Comment:
It would be interesting to time this, vs checking `applicable` vs Tricks.jl's `static_applicable`

Reply:
I'm not sure it's a bit priority for our use-case, but I've made an issue to look into it more. Might provide minor improvements for specific types.

https://github.com/invenia/Impute.jl/issues/82